test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1

```diff
@@ -73,6 +73,32 @@ Describe "Get-Content" -Tags "CI" {
     It "should throw 'PSNotSupportedException' when you set-content to an unsupported provider" -Skip:($IsLinux -Or $IsMacOS) {
         {get-content -path HKLM:\\software\\microsoft -ea stop} | Should Throw "IContentCmdletProvider interface is not implemented"
     }
+    It 'Verifies -Tail reports a TailNotSupported error for unsupported providers' {
+        {Get-Content -Path Variable:\PSHOME -Tail 1 -ErrorAction Stop} | ShouldBeErrorId 'TailNotSupported,Microsoft.PowerShell.Commands.GetContentCommand'
+    }
+    It 'Verifies using -Tail and -TotalCount together reports a TailAndHeadCannotCoexist error' {
+            { Get-Content -Path Variable:\PSHOME -Tail 1 -TotalCount 5 -ErrorAction Stop} | ShouldBeErrorId 'TailAndHeadCannotCoexist,Microsoft.PowerShell.Commands.GetContentCommand'
+    }
+    It 'Verifies -Tail with content that uses an uncommon encoding' {
```

Comment:
Should we use -TestCases and validate with more than one encoding?

Reply:
Yes; fixed.